jdaviz/tests/test_app.py

```diff
@@ -141,3 +141,24 @@ def test_case_that_used_to_break_return_label(specviz_helper, spectrum1d):
     dc = specviz_helper.app.data_collection
     assert dc[0].label == "this used to break (1)"
     assert dc[1].label == "this used to break (2)"
+
+
+def test_viewer_renaming(specviz_helper, imviz_helper):
+    viewer_names = [
+        'spectrum-viewer',
+        'second-viewer-name',
+        'third-viewer-name'
+    ]
+
+    for i in range(len(viewer_names) - 1):
+        specviz_helper.app._update_viewer_reference_name(
+            old_reference=viewer_names[i],
+            new_reference=viewer_names[i + 1]
+        )
+        assert specviz_helper.app.get_viewer(viewer_names[i+1]) is not None
+
+    with pytest.raises(ValueError):

```

Comment:
Maybe also check the message with `match=` just to be sure it is the ValueError that you expected.